users/models.py

```diff
@@ -1,3 +1,385 @@
 from django.db import models
+from django.contrib.auth.models import AbstractBaseUser, BaseUserManager, PermissionsMixin
```

Comment:
You are importing both AbstractBaseUser and PermissionsMixin, but also importing Group and Permission for custom relations in the User model. Make sure you really need to override the default relations - Django's built-in fields already provide this. If not, you can use the defaults for simpler maintenance.